### PR TITLE
Minor updates and improvements to the notebooks

### PR DESCRIPTION
**Pull request by fmassa**
This PR contains some minor updates to the current notebooks.
The additions generally include using some newer PyTorch constructs, plus some minor stylistic changes (like using `@torch.no_grad()` as a decorator when the whole function is meant to be on non-grad mode).
Feel free to cherry-pick particular changes, if some of them you do not necessarily agree with. I've removed all the `!pip3 install torch` because on colab PyTorch is now installed by default (and I think the same applies to SageMaker?)

---

There is something else that I'd like to comment on the `Seedling` exercise that I didn't change here.
When you setup the dataset, you create the `full_dataset` and then split it in two via `torch.utils.data.random_split`. This generally works fine, but one thing to keep in mind is that there is a random flip in your original dataset. This means that the validation dataset will also contain this randomness factor, which is generally not desired.

One way to fix this would be to pass a `train=True` flag to the `get_transform`, and create two datasets: 
```python
full_dataset_train = torchvision.datasets.ImageFolder('train', transform=get_transforms(train=True))
full_dataset_val = torchvision.datasets.ImageFolder('train', transform=get_transforms(train=False))
```
and when selecting the subset of the dataset, using `torch.utils.data.Subset` instead:
```python
train_len = int(0.8 * len(full_dataset_train))
validate_len = len(full_dataset_train) - train_len
idxs = torch.randperm(len(full_dataset_train)).tolist()
idxs_train, idxs_val = idxs.split([train_len, validate_len], dim=0)
train_dataset = torch.utils.data.Subset(full_dataset_train, idxs_train)
val_dataset = torch.utils.data.Subset(full_dataset_val, idxs_val)
```

I didn't want to change this because it's a slightly bigger change, and I'd like you go check it out before.

### CODE REVIEW

File: PyTorch Sentiment Analysis Exercise.ipynb [hunk 1]
```diff
@@ -301,9 +301,7 @@
     "        super(SentimentAnalyzer, self).__init__()\n",
     "        self.hidden_size = hidden_size\n",
     "        \n",
-    "        self.embedding = nn.Embedding(vocab_size, embedding_size)\n",
-    "        self.embedding.weight.data.copy_(pretrained_embedding)\n",
-    "        self.embedding.weight.requires_grad = False\n",
+    "        self.embedding = nn.Embedding.from_pretrained(pretrained_embedding, freeze=True)\n",
```
Comment (fmassa) [inline PyTorch Sentiment Analysis Exercise.ipynb:304]: Now that I've changed this to use `from_pretrained`, we don't need `vocab_size` in this class anymore. Feel free to remove it if you prefer to keep this change, or let me know if you want me to change this

File: cifar10_tutorial.ipynb [hunk 2]
```diff
@@ -207,7 +207,7 @@
     "    def forward(self, x):\n",
     "        x = F.max_pool2d(F.relu(self.conv1(x)), 2)\n",
     "        x = F.max_pool2d(F.relu(self.conv2(x)), 2)\n",
-    "        x = x.view(-1, 16 * 5 * 5)\n",
+    "        x = x.flatten(start_dim=1)\n",
```
Comment (fmassa) [inline cifar10_tutorial.ipynb:210]: I think for now on, the recommended approach for this kind of flattening is to use `.flatten`, as it is simpler to use and fits well with the use-cases that we have for going from convolutional feature maps to 2d tensors

File: transfer_learning.ipynb [hunk 3]
```diff
@@ -317,13 +294,14 @@
     "                    if phase == 'train':\n",
     "                        loss.backward()\n",
     "                        optimizer.step()\n",
+    "                        scheduler.step()\n",
```
Comment (fmassa) [inline transfer_learning.ipynb:297]: There was a change in PyTorch 1.1 that now makes the scheduler go after the first `optimizer.step()`, see https://github.com/pytorch/pytorch/pull/7889

File: transfer_learning.ipynb [hunk 4]
```diff
@@ -372,31 +350,31 @@
...
+    "\n",
+    "        outputs = model(inputs)\n",
+    "        _, preds = torch.max(outputs, 1)\n",
+    "\n",
+    "        for j in range(inputs.size(0)):\n",
+    "            images_so_far += 1\n",
+    "            ax = plt.subplot(num_images // 2, 2, images_so_far)\n",
+    "            ax.axis('off')\n",
+    "            ax.set_title('predicted: {}'.format(class_names[preds[j]]))\n",
+    "            imshow(inputs[j])\n",
```
Comment (fmassa) [inline transfer_learning.ipynb:372]: Everything in this block is the same, except this line, where I changed `inputs.cpu().data[j]` to `inputs[j]`. This works because in `imshow` I added `.detach().cpu()` to make it slightly more robust

File: transfer_learning.ipynb [hunk 5]
```diff
@@ -289,11 +270,7 @@
     "\n",
     "        # Each epoch has a training and validation phase\n",
     "        for phase in ['train', 'val']:\n",
-    "            if phase == 'train':\n",
-    "                scheduler.step()\n",
-    "                model.train()  # Set model to training mode\n",
-    "            else:\n",
-    "                model.eval()   # Set model to evaluate mode\n",
+    "            model.train(phase == 'train')  # Set model to training / eval mode\n",
```
Comment (fmassa) [inline transfer_learning.ipynb:273]: Feel free to revert this change if you think it was clearer before (but remove the `scheduler.step()`, as I mentioned in the other comment

File: transfer_learning.ipynb (hunk 3 above)
Comment (fmassa) [reply transfer_learning.ipynb:297]: annnd, this introduced a bug :-)

It should happen once per epoch, and not once every iteration. Sorry about the bug!
